From bc852785b678607fdaff0bcfe6a63995b54e4c0e Mon Sep 17 00:00:00 2001
From: Igor Mammedov <imammedo@redhat.com>
Date: Wed, 17 Sep 2014 12:52:29 -0400
Subject: [x86] smpboot: Log error on secondary CPU wakeup failure at ERR level

Message-id: <1410958350-13432-4-git-send-email-imammedo@redhat.com>
Patchwork-id: 95867
O-Subject: [RHEL7.1 PATCH 3/4] x86/smpboot: Log error on secondary CPU wakeup failure at ERR level
Bugzilla: 968147
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=968147
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7981610
Upstream: feef1e8ecbadf24f8e6829c935df8683cabae41b

If system is running without debug level logging,
it will not log error if do_boot_cpu() failed to
wakeup AP. It may lead to silent AP bringup
failures at boot time.
Change message level to KERN_ERR to make error
visible to user as it's done on other architectures.

Signed-off-by: Igor Mammedov <imammedo@redhat.com>
Acked-by: Toshi Kani <toshi.kani@hp.com>
Cc: Thomas Gleixner <tglx@linutronix.de>
Link: http://lkml.kernel.org/r/1401975765-22328-3-git-send-email-imammedo@redhat.com
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index e210ab7..9ba8ede 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -898,7 +898,7 @@ int native_cpu_up(unsigned int cpu, struct task_struct *tidle)
 
  err = do_boot_cpu(apicid, cpu, tidle);
  if (err) {
-  pr_debug("do_boot_cpu failed %d\n", err);
+  pr_err("do_boot_cpu failed(%d) to wakeup CPU#%u\n", err, cpu);
   return -EIO;
  }
 
-- 
1.7.1